src/pool/disjoint/pool_disjoint_heap_manager.cpp

```diff
@@ -0,0 +1,112 @@
+// Copyright (C) 2023 Intel Corporation
+// Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <umf/memory_pool_ops.h>
+#include <umf/memory_provider.h>
+
+#include "pool/pool_disjoint.h"
+#include "pool_disjoint.hpp"
+
+struct disjoint_memory_pool {
+    umf_memory_provider_handle_t mem_provider;
+    usm::DisjointPool *disjoint_pool;
+};
+
+enum umf_result_t
+disjoint_pool_initialize(umf_memory_provider_handle_t *providers,
+                         size_t numProviders, void *params, void **pool) {
```

Comment:
Since you are using `new` inside the function you should wrap the function with `try/catch` and translate the exception to a umf error: you can put try catch around the entire function like this:

```
disjoint_pool_initialize(umf_memory_provider_handle_t *providers,
                         size_t numProviders, void *params, void **pool)  try {
  ...
} catch(std::bad_alloc&) {
   return UMF_RESULT_ERROR_OUT_OF_HOST_MEMORY;
} catch (...) {
   return UMF_RESULT_UNKNOWN_ERROR:
}

```

Reply:
Done: https://github.com/oneapi-src/unified-memory-framework/pull/17